app/helpers/curation_concerns/collections_helper.rb

```diff
@@ -17,8 +17,9 @@ def link_to_select_collection(collectible, opts = {})
 
   # override hydra-collections
   def link_to_remove_from_collection(document, label = 'Remove From Collection')
-    link_to collections.collection_path(@collection.id, collection: { members: 'remove' },
-                                                        batch_document_ids: [document.id]), method: :put do
+    collection_id = @collection ? @collection.id : @presenter.id
```

Comment:
The show page uses the collection presenter and the edit page use the actual collection object. If there is a collection instance variable its id gets passed, otherwise the presenter's id gets passed.